sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/DataSourceCredentialEntity.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.AI.MetricsAdvisor.Models
+{
+    /// <summary>
+    /// The credential used to authenticate a <see cref="DataFeedSource"/> for data ingestion.
```

Comment:
The credential entity will be passed to a `DataFeed` for authentication, but this is not implemented in this PR yet. We'll first get all CRUD operations of the new API implemented and tested before that.